### ISSUE TITLE
IDISP007 false positive

### ISSUE THREAD

**Issue by sungam3r** [DotNetAnalyzers/IDisposableAnalyzers#236]
```c#
_connectionFactory.CreateConnection()?.Dispose();
```

gives `IDISP007	Don't dispose injected.`,  

while 
```c#
 var c = _connectionFactory.CreateConnection();
 c?.Dispose();
```
not (right behaviour). `_connectionFactory` **is** `IDisposable` indeed but it is not disposed, the `connection` object from `CreateConnection()` is one being disposed.

**Comment by JohanLarsson** [comment]
Looks like a dumb bug, sorry about it, thanks for reporting!

**Comment by JohanLarsson** [comment]
Fixed in 3.4.3

**Comment by sungam3r** [comment]
Thanks. It works.

**Comment by JohanLarsson** [comment]
Keep the issues coming, I don't use web APIs much so this project relies on issues for those.
